message/message-util/util/src/java/org/sakaiproject/message/util/BaseMessage.java

```diff
@@ -1961,6 +1991,25 @@ else if (element2.getTagName().equals(SYNOPTIC_TOOL))
 
 	} // merge
 
+	/**
+	 * Extract the tool name from an attachment URL path - Can be overridden in the extending classes
+	 * @param url The attachment URL path
+	 * @return The tool name (e.g. "Announcements")
+	 */
+	public String getToolTitle(String url) {
+		if (url == null) return "";
+
+		// Split path on "/" and look for tool name after "attachment" segment
+		// /content/attachment/a54ab888-26d3-43db-bd7e-ff97b6192a76/Announcements/495639a1-dbaf-4855-af0c-9a51dc6db50a/ietf-jon-postel-02.png
+		String[] parts = url.split("/");
+		for (int i = 0; i < parts.length - 1; i++) {
```

Comment:
Maybe this should be more elegant using Patter and Matcher

Reply:
Done.